addons/mail/models/mail_link_preview_message.py

```diff
@@ -0,0 +1,45 @@
+from odoo import fields, models
+from odoo.addons.mail.tools.discuss import Store
+
+
+class MailLinkPreviewMessage(models.Model):
+    _name = "mail.link.preview.message"
+    _inherit = ["bus.listener.mixin"]
+    _description = "Link between link previews and messages"
+
+    link_preview_id = fields.Many2one("mail.link.preview", index=True, required=True, ondelete="cascade")
+    message_id = fields.Many2one("mail.message")
+    is_hidden = fields.Boolean()
+
+    _unique = models.UniqueIndex("(message_id, link_preview_id)")
+
+    def _bus_channel(self):
+        return self.message_id._bus_channel()
+
+    def _hide_and_notify(self):
+        if not self:
+            return True
+        self.is_hidden = True
+        for link_preview_message in self:
+            link_preview_message._bus_send_store(link_preview_message)
+
+    def _unlink_and_notify(self):
+        if not self:
+            return True
+        for link_preview_message in self:
+            link_preview_message._bus_send_store(link_preview_message)
+        self.unlink()
+
+    def _to_store(
+        self,
+        store: Store,
+        /,
+    ):
```

Comment:
Don't use lines when not necessary (not like this definition is complicated to read, especially with a brain freed of useless information about orm) :
```suggestion
    def _to_store(self, store: Store, /):
```